packages/login_client/lib/src/login_client.dart

```diff
@@ -87,8 +83,13 @@ class LoginClient extends http.BaseClient {
   Future<oauth2.Credentials?> get credentials => _credentialsStorage.read();
 
   /// Restores saved credentials from the credentials storage.
-  Future<void> initialize() async {
-    final credentials = await _credentialsStorage.read();
+  ///
+  /// You can provide a custom `tokenEndpoint` to override the one stored in
+  /// the credentials. This is useful when the token endpoint changes.
+  Future<void> initialize({Uri? tokenEndpoint}) async {
```

Comment:
I wouldn't accept the `tokenEndpoint` from the outside but rather a flag that would change the source of the token endpoint for refreshing it or I would change the behavior altogether (without a flag). The `tokenEndpoint` used shall come from `_oAuthSettings.authorizationUri`.

@lukaszgarstecki is it safe to make such a change to our client, that always the endpoint passed to the OAuth client settings is used for refreshing the token instead of trying to use the same endpoint that was used for the initial authorization? From the top of my head, I can't think of circumstances that would make it an unwanted behavior.

Reply:
After discussion we decided to go with the second approach (without flag). By default we always set for `tokenEndpoint` value taken from `OAuthSettings.authorizationUri`